crates/networking/p2p/types.rs

```diff
@@ -267,8 +267,12 @@ impl NodeRecord {
                     decoded_pairs.secp256k1 = Some(H264::from_slice(&bytes))
                 }
                 "eth" => {
-                    // the first byte is ignored as an array within an array is received
-                    decoded_pairs.eth = ForkId::decode(&value[1..]).ok();
+                    let Ok(decoder) = Decoder::new(&value) else {
+                        continue;
+                    };
+                    let (fork_id, decoder) = decoder.decode_optional_field();
+                    decoder.finish_unchecked();
+                    decoded_pairs.eth = fork_id;
```

Comment:
Im confused, there is not reference to any list here, whereas the description of the PR is refering to a list of forks.

Reply:
The decoder allows decoding an RLP list of different types (mostly to build a struct out of them). I added a comment that explains a bit what we're doing.